shellbase-slack/src/main/scala/com/sumologic/shellbase/slack/PostToSlackHelper.scala

```diff
@@ -29,12 +31,15 @@ trait PostToSlackHelper {
 
   def slackMessagingConfigured = slackState.slackClient.isDefined && slackState.slackChannel.isDefined
 
-  def sendSlackMessageIfConfigured(msg: String, additionalOptions: Map[String, String] = Map.empty): Unit = {
+  def sendSlackMessageIfConfigured(msg: String, additionalOptions: Map[String, String] = Map.empty):
+  Option[PostMessageResponse] = {
+    var message : Option[PostMessageResponse] = None
     if (!blacklistedUsernames.contains(username)) {
       for (client <- slackState.slackClient;
            channel <- slackState.slackChannel) {
-        client.chat.postMessage(channel, msg, slackState.slackOptions ++ additionalOptions)
+        message = Some(client.chat.postMessage(channel, msg, slackState.slackOptions ++ additionalOptions))
```

Comment:
Let's not assume this returns non-null.  It's strictly better to defend with `Option` IMO.